### PR TITLE
Update the SDK to the stable Protocol 20 release

### LINKED ISSUE #552: Updating the SDKs to the stable Protocol 20 release
SDKs need to be updated to adapt to breaking changes that have occurred since Preview 11 (i.e. the current testnet release candidate) coming to Protocol 20.

## Summary of Breaking Changes
This is likely irrelevant to SDK maintainers, but do note sandbox mode has been removed from the Soroban CLI (https://github.com/stellar/soroban-tools/issues/983).

### XDR
The XDR needs to be updated to https://github.com/stellar/stellar-xdr/commit/6a620d160aab22609c982d54578ff6a63bfcdc01. This includes a non-trivial list of breaking changes (mostly renames):
  - The `BumpFootprintExpirationOp` is now `ExtendFootprintTTLOp` and its `ledgersToExpire` field is now named `extendTo`, but it serves the same purpose.
  - `ContractExecutable.contractExecutableToken` is now `contractExecutableStellarAsset`
  - `SorobanTransactionData.refundableFee` is now `resourceFee`
  - This new fee encompasses the entirety of the Soroban-related resource fees. Note that this is distinct from the "network-inclusion" fee that you would set on your transaction
  - see https://github.com/stellar/js-stellar-base/pull/704 for an example migration

### RPC
 - The `getEvents` RPC endpoint structure has changed:
   * `value: { xdr: string }` is now just `value: string` (https://github.com/stellar/soroban-tools/pull/1015)
   * `contractId` will always be a contract strkey (`C...` format) if present (https://github.com/stellar/soroban-tools/issues/1007)
 - The `getLedgerEntries` RPC endpoint now includes expiration info (https://github.com/stellar/soroban-tools/pull/1010):
   * `expirationLedgerSeq` is a string-ified `uint32` present if the requested entries have expiration

### PR DESCRIPTION
**Pull request by overcat**
Closes #552

### CODE REVIEW

File: src/main/java/org/stellar/sdk/responses/operations/ExtendFootprintTTLOperationResponse.java [hunk 1]
```diff
@@ -3,7 +3,9 @@
 import com.google.gson.annotations.SerializedName;
 
 /**
- * Represents BumpFootprintExpiration operation response.
+ * Represents ExtendFootprintTTL operation response.
```
Comment (overcat) [inline src/main/java/org/stellar/sdk/responses/operations/ExtendFootprintTTLOperationResponse.java:6]: Currently, the code for Horizon has not been updated, so we need to update this link later.

File: src/main/java/org/stellar/sdk/responses/operations/ExtendFootprintTTLOperationResponse.java [hunk 2]
```diff
@@ -3,7 +3,9 @@
...
+ *
+ * <p>TODO: update link
```
Comment (overcat) [inline src/main/java/org/stellar/sdk/responses/operations/ExtendFootprintTTLOperationResponse.java:8]: Currently, the code for Horizon has not been updated, so we need to update this link later.

**Comment by overcat** [comment]
Hi @sreuland, can you please review this PR?

File: src/main/java/org/stellar/sdk/responses/operations/ExtendFootprintTTLOperationResponse.java (hunk 2 above)
Comment (sreuland) [reply src/main/java/org/stellar/sdk/responses/operations/ExtendFootprintTTLOperationResponse.java:8]: ok, https://github.com/stellar/go/issues/5089 is pending for update on horizon

**Review by sreuland — APPROVED** [review]
nice work, incorporating the xdr changes throughout.

**Pull request merged**